### ISSUE TITLE
`core::cmp::PartialEq` and `core::cmp::PartialOrd` are flagged as `#[const_trait]` but aren't constant traits 

### ISSUE THREAD

**Issue by tadaHrd** [rust-lang/rust#112926]
Here's how the core source code of the traits in textual form looks like:
```rust
…
#[const_trait]
…
pub trait PartialEq<Rhs: ?Sized = Self> {
    …
}
…
#[const_trait]
…
pub trait PartialOrd<Rhs: ?Sized = Self>: PartialEq<Rhs> {
    …
}
```

I got this error:
```
error: ~const can only be applied to `#[const_trait]` traits
 --> src\structs\comptime_map.rs:5:16
  |
5 | impl<K: ~const PartialEq, V, const L: usize> …
  |                ^^^^^^^^^
```

It should not happen because the trait is flagged as constant in the textual form of the `core` library.

**Comment by tadaHrd** [comment]
Thank you for your knowledge

> These traits are not marked as const on nightly.
